### PR TITLE
[Writing Code] Add learning objectives

### PR DESCRIPTION
**Pull request by ehumph**
This PR adds learning objectives to the chapter on Writing Code.

### CODE REVIEW

File: 03-writing_code.Rmd [hunk 1]
```diff
@@ -6,6 +6,16 @@ ottrpal::set_knitr_image_path()
 
 # Writing Code
 
+## Learning Objectives
+
+- Recognize the benefits and capabilities of AI-powered programming tools
+- Explain how AI-generated code and automated programming tools contribute to making programming more accessible to individuals with limited coding experience
+- Identify the evolving skills required for programming with AI
```
Comment (carriewright11) [inline 03-writing_code.Rmd:13]: Wonder if we could expand this a bit more - is this about having specific prompts?

File: 03-writing_code.Rmd [hunk 2]
```diff
@@ -6,6 +6,16 @@ ottrpal::set_knitr_image_path()
...
+- Explore the ethical considerations associated with using AI in programming, including the importance of rigorous testing before publishing code
+- Apply strategies for using AI to build functions, debug code, understand error messages, choose the proper programming language, and plan out a multi-step programming project.
```
Comment (carriewright11) [inline 03-writing_code.Rmd:15]: maybe add checking syntax?

File: 03-writing_code.Rmd (hunk 1 above)
Comment (carriewright11) [reply 03-writing_code.Rmd:13]: I get it though if you don't want to give that away just yet

Comment (ehumph) [reply 03-writing_code.Rmd:13]: I ended up taking this objective out. The chapter spends only a couple of sentences on the topic and I think having it as an objective is misleading.

**Comment by ehumph** [comment]
> Looking good! I am realizing that it might be nice to address (maybe by adding it to sec 3.1) the fact that AI can be useful for writing in languages that you are less familiar with even if you are an experienced programmer? Just thinking that the learning objectives and the list in 3.1 really emphasize use by people who have limited coding experience, but I don't think we want to leave out the other potential audience. I also kinda think that people using it too much with limited experience could still have some negative consequences currently given that it still makes a lot of mistakes.

I added some language addressing this in 3.1

File: 03-writing_code.Rmd (hunk 2 above)
Comment (ehumph) [reply 03-writing_code.Rmd:15]: Added!

File: 03-writing_code.Rmd [hunk 3]
```diff
@@ -6,6 +6,15 @@ ottrpal::set_knitr_image_path()
 
 # Writing Code
 
+## Learning Objectives
+
+- Recognize the benefits and capabilities of AI-powered programming tools, particularly for certain kinds of tasks that are difficult for humans
+- Explain how AI-generated code and automated programming tools contribute to making programming more accessible to individuals with limited coding experience
```
Comment (carriewright11) [inline 03-writing_code.Rmd:12]:
```suggestion
- Explain how AI-generated code and automated programming tools contribute to making programming more accessible to individuals with limited coding experience or to those working in a new language
```

Comment (carriewright11) [reply 03-writing_code.Rmd:12]: How's this? just to emphasize that point a tad more

**Review by carriewright11 — COMMENTED** [review]
Besides my one suggestion (which is just a suggestion) it LGTM!

**Pull request merged**